### PR TITLE
Support list type in analyze

### PR DESCRIPTION
**Pull request by hanxiaop**
**Please provide a description of this PR:**
Fix https://github.com/istio/istio/issues/39982

### CODE REVIEW

**Review by GregHanson — COMMENTED** [review]
Can you add a release note with a linked issue?

File: pilot/pkg/config/file/store.go [hunk 1]
```diff
@@ -282,12 +282,26 @@ func (s *KubeSource) parseContent(r *collection.Schemas, name, yamlText string)
 		}
 
 		chunk := bytes.TrimSpace(doc)
-		r, err := s.parseChunk(r, name, lineNum, chunk)
+		resource, err := s.parseChunk(r, name, lineNum, chunk)
```
Comment (howardjohn) [inline pilot/pkg/config/file/store.go:285]: I would justchange parseChunk to natively handle List, parsing error feels a bit problematic?

Comment (hanxiaop) [reply pilot/pkg/config/file/store.go]: @howardjohn Done. Thanks

**Review by GregHanson — APPROVED** [review]

File: pilot/pkg/config/file/store.go [hunk 2]
```diff
@@ -400,15 +415,59 @@ func (s *KubeSource) parseChunk(r *collection.Schemas, name string, lineNum int,
...
+	}
+	yamlNode := yamlChunkNode.Content[0]
+	var itemsInd int
+	for ; itemsInd < len(yamlNode.Content); itemsInd++ {
+		if yamlNode.Content[itemsInd].Kind == yamlv3.ScalarNode && yamlNode.Content[itemsInd].Value == "items" {
+			itemsInd++
+			break
+		}
+	}
+	if itemsInd >= len(yamlNode.Content) && yamlNode.Content[itemsInd].Kind != yamlv3.SequenceNode {
```
Comment (howardjohn) [inline pilot/pkg/config/file/store.go:452]: if `itemsInd >= len` won't `yamlNode.Content[itemsInd]` be out of bounds?

@hanxiaop

File: pilot/pkg/config/file/store.go [hunk 3]
```diff
@@ -282,11 +282,10 @@ func (s *KubeSource) parseContent(r *collection.Schemas, name, yamlText string)
 		}
 
 		chunk := bytes.TrimSpace(doc)
-		r, err := s.parseChunk(r, name, lineNum, chunk)
+		chunkResources, err := s.parseChunk(r, name, lineNum, chunk)
```
Comment (WM7586) [inline pilot/pkg/config/file/store.go:285]:
```suggestion
		chunkResources, err := s.parseChunk(r, name, lineNum, chunk)
```suggestion
		chunkResources, err := s.parseChunk(r, name, lineNum, chunk)_[]()[]()![git-usage-clone-ui.png](https://user-images.githubusercontent.com/117557187/220818925-b6bafe43-ee9e-43f1-966a-e806de0d8b69.png)

<!-- Error while uploading d6e875ee-692e-41eb-a80b-f42ad7d88058 -->![git-usage-git-status-untracked.png](https://user-images.githubusercontent.com/117557187/22081![git-usage-clone-ui.png](https://user-images.githubusercontent.com/117557187/220818933-a3844634-9546-4fff-be73-a52d557b3b8b.png)

ages.githubusercontent.com/117557187/220818914-8c0d726a-b45a-470c-b3f7-1ea89010de76.png)

.png)

![All About That Bass.jpg](https://user-images.githubusercontent.com/117557187/2208![git-usage-git-status-added.png](https://user-images.githubusercontent.com/117557187/220818917-f3865d96-33fc-4b6d-9a65-193e2564f270.png)

![Uploading git-usage-clone-ui.png …]()

![git-usage-git-status-added.png](https://user-images.githubusercontent.com/117557187/220818930-e3d38a76-b3ab-4bc4-afc1-a639932ded8d.png)

![git-usage-git-status-clean.png](https://user-images.githubusercontent.com/117557187/220818928-a9663c![Copia di Tutti i brani.m3u-1.jpg](https://user-images.githubusercontent.com/117557187/220818916-50a33200-2a7f-411a-8472-e3d733df1413.jpg)

ages.githubusercontent.com/117557187/220818921-d691bff8-d150-4ae0-b939-a45e17593eb4.jpg)

ages.githubusercontent.com/117557187/220818929-4e43335b-a883-4706-a789-ee59881a0617.jpg)

ti i brani.m3u (2).jpg](https://user-images.githubusercontent.com/117557187/220818919-0ee567b8-17ca-484c-b965-f600126bf1bb.jpg)

![U![git-usage-git-status-untracked.png](https://user-images.githubusercontent.com/117557187/220818924-baf7fe18-4cba-402f-981d-501bd661ee18.png)

-usage-push-ui.png](https://user-images.githubusercontent.com/117557187/220818920-46b9daea-d3b5-426c-9a51-034eb0e868dc.png)

![Uploading Copia di Tutti i brani.m3u-1.jpg …]()

![Uploading git-usage-clone-ui.png …]()

![Uploading git-usage-git-status-untracked.png …]()

![Copia di Tutti i brani.m3u (1).jpg](https://user-images.githubusercontent.com/117557187/220818923-e8bf630b-2d![ic_tool_box_quick_setting.png](https://u![git-usage-push-ui.png](https://user-images.githubusercontent.com/117557187/220818927-62c28739-e872-4e5c-9335-7a08ddb0ac06.png)

87/220818932-0a20f7d5-fe18-4b77-8b79-d47b8861fbff.png)

usage-git-status-clean.png …]()

![Uploading Copia di Tutti i brani.m3u-1.jpg …]()

![Uploading git-usage-push-ui.png …]()

![Uploading git-usage-git-status-added.png …]()

![Copia di Tutti i br![1620153153327.jpg](https://user-images.githubusercontent.com/117557187/220818935-f83f7779-4534-4315-ac2b-308bb1809ea9.jpg)

hubusercontent.com/117557187/220818931-b0e2fb33-98e4-46b6-8851-81d1cde1c6dc.jpg)

![Uploading ic_tool_box_quick_setting.png …]()

![Screenshot_20230210-192625.jpeg](https://user-images.githubusercontent.com/117557187/220818934-ceee98a7-223e-4041-a62f-c1dee6fea9d9.jpeg)

<!-- Error while uploading VID_20230202_061007.3gp -->![Uploading 1620153153327.jpg …]()

![Uploading muhammed-fardeen-finos-790rDizAjl4-un<!-- Error while uploading 4f34f941-5eda-490c-a850-cc4b1bd0d4eb -->4b1bd0d4eb …]()

![IMG_20230202_061020.jpg](https://user-images.githubusercontent.com/117557187/220818937-21ec0f86-3cef-4a39-83b4-9b263fb4ae90.jpg)

![IMG_20230202_060954.jpg](https://user-images.githubusercontent.com/117557187/220818938-13bc6f5c-b57a-49b1-8d1c-2427d9954f55.jpg)

- [ ] _
```

```

**Pull request merged**